packages/components/src/popover/index.tsx

```diff
@@ -499,8 +499,14 @@ const UnforwardedPopover = (
 							// to use `translateX` and `translateY` because those values would
 							// be overridden by the return value of the
 							// `placementToMotionAnimationProps` function in `AnimatedWrapper`
-							x: Math.round( x ?? 0 ) || undefined,
-							y: Math.round( y ?? 0 ) || undefined,
+							x:
+								x === null || Number.isNaN( x )
+									? 0
```

Comment:
On second thoughts, perhaps this side of the ternary should be `undefined` as prior to this PR, the `||` meant that nullish or `NaN` values resulted in `undefined`... I can give that a try tomorrow, as that might potentially resolve the issue with the failing test fixtures.

Reply:
Thanks for outlining each of the different options. To try to be more conservative about it, I've updated this logic in this PR to `c === null || Number.isNaN( c ) ? undefined : Math.round( c )` so that we're only outputting a number when we have a real value. I think that might get us to a smaller potential change? Happy to revert if need be.

Reply:
That looks good to me. Looking at the snapshot changes, it's interesting that motion stops applying a transform on the `Z` axis — it must be some internal framer motion optimisation. From past experience, usually the reason for having `translateZ(0)` would be to promote an element to a separate composite layer, thus reducing the amount of times it needs to be re-painted to screen. But I can see that we're already applying the `will-change: transform` style, which also achieves the same goal, so we should also be fine under that point of view.